mender-dist-packages-building/install-armv6-cross-toolchain

```diff
@@ -0,0 +1,170 @@
+    libboost-system1.67.0
+    libboost-thread-dev
+    libboost-thread1.67-dev
+    libboost-thread1.67.0
+    libboost1.67-dev
+    libbz2-1.0
+    libgcc1
+    libgomp1
+    libicu-dev
+    libicu63
+    libidn2-0
+    liblmdb-dev
+    liblmdb0
+    liblz4-1
+    liblzma5
+    libmount1
+    libnettle6
+    libpcre3
+    libselinux1
+    libssl-dev
+    libssl1.1
+    libxml2
+    libzstd1
+)
+
+function cleanup() {
+    rm -f Packages data.tar.xz *.deb;
+}
+
+# trap cleanup EXIT
+
+function download_pkg() {
+    echo "Downloading ${pkg}...";
+    local -r armv6_toolchain_root=$1
+    local -r pkg=$2
+    deb_package_url=$(grep Filename Packages | grep /${pkg}_ | grep armhf | tail -n1 | sed 's/Filename: //');
+    filename=$(basename $deb_package_url);
+    curl --location http://raspbian.raspberrypi.org/raspbian/${deb_package_url} -o $filename 2>/dev/null || { echo "Did not find the package: ${pkg} in the raspbian repos.."; exit 1; }
+};
+
+function install_cross_compiler_toolchain() {
+    # Install the armv6 supported cross toolchain from Kotlin
+    [[ $# -ne 3 ]] && { echo "Three arguments required. got $#"; exit 1; }
+    local -r install_prefix="$1"
+    local -r download_url="$2"
+    local -r toolchain_version="$3"
+    echo "Installing the cross compiler toolchain to ${install_prefix}<cross-compiler>"
+    (
+        cd "$install_prefix"
+        wget -nc -q ${download_url}${toolchain_version}.tar.bz2
+        # wget -nc -q https://toolchains.bootlin.com/downloads/releases/toolchains/armv6-eabihf/tarballs/armv6-eabihf--glibc--stable-2020.08-1.tar.bz2
+        tar -xjf ${toolchain_version}.tar.bz2
+        # tar -xjf armv6-eabihf--glibc--stable-2020.08-1.tar.bz2
+        rm ${toolchain_version}.tar.bz2;
+        # rm armv6-eabihf--glibc--stable-2020.08-1.tar.bz2;
+    )
+}
+
+# Get depdendencies from upstream, manually donwloading deb packages, and fake pkg-config.
+# NOTE: pkg-config is used from go build to obtain cflags and libs required to build C code; however
+# for armhf we will explicitly pass these flags for go to use our custom sysroot instead of the system
+# one. Alternatively pkg-config supports setting a prefix, but that would require patching the source code.
+function setup_pkg_config() {
+    local -r URL="$1"
+    ln -s /bin/true /usr/bin/pkg-config
+    curl -f "${URL}" -o Packages.gz
+    gunzip Packages.gz
+}
+
+function install_packages_to_toolchain_root() {
+    local -r toolchain_root="$1"; shift
+    local -r required_packages_="$@"
+    for pkg in ${required_packages_} ; do
+        download_pkg ${toolchain_root} ${pkg}
+        echo "Extracting ${pkg} to an archive (tar.(xz|gz))...";
+        ar -x ${pkg}_*_armhf.deb data.tar.xz;
+        for dir in ./lib ./usr/include ./usr/lib; do
+            tar --list --file=data.tar.xz ${dir} >& /dev/null && tar --extract  --directory ${toolchain_root}/arm-buildroot-linux-gnueabihf/sysroot/ --file=data.tar.xz --transform='s,/arm-linux-gnueabihf,,'  ${dir} || true
```

Comment:
:raised_hands:

Reply:
I'm thinking I should probably add a comment here, about stripping the cross-compiler path?

Reply:
Up to you. Not a big deal from my side